### PR TITLE
build(deps): AUT-1625 bump transformers to 5.15.1

### PR DESCRIPTION
**Pull request by svcnemo-autobot**
## Background

Transformers 5.15.1 introduces heterogeneous per-layer configs and renamed model, generation, speculative-decoding, and tokenizer APIs.

## What changed

- Pin Transformers 5.15.1 and regenerate lockfiles.
- Adapt Gemma4, Nemotron, Qwen3.5, Inkling, DSpark, and EAGLE integrations.
- Update the copied Mistral tokenizer and affected fixtures.
- Merge current `main`, including DeepSeek V4 PP/EP parity coverage from #3613.

## Details

```mermaid
flowchart LR
  T[Transformers 5.15.1] --> C[compatibility adapters]
  C --> M[model APIs]
  C --> P[distributed paths]
  C --> K[tokenization]
```

## Tested

- Exact-head lock, lint, type, build, and GitHub L2 lanes, including VLM, parallelism, context-parallel, and MoE, are terminal success/skipped in the latest read; the aggregate verdict remains pending.
- DeepSeek V4 PP2 and EP2 parity passed on 2 H100s, comparing six loss and gradient-norm steps each.
- Nemotron V3 hybrid CP passed all six parity configurations on 2 H100s.
- The release-resolved Nemotron V3 CP1/EP8 and CP2/EP4 recipe passed 50 steps per leg on 8 H100s and 50-step loss/gradient parity.
- Gemma4, Qwen3.5 VLM/MoE, and Inkling release recipes remain blocked by unavailable model/MedPix fixtures and, for larger shapes, the 32-GPU session ceiling.

### CODE REVIEW

**Comment by ko3n1g** [comment]
/ok to test [5a3f103](https://github.com/NVIDIA-NeMo/Automodel/pull/3615/commits/5a3f1033999a046e19c01380500a1728c0d5955f)

**Comment by athitten** [comment]
Could we run the GitLab nemo-ci release suite before merging this?

GitHub CI is green, but it caps at 2 GPUs, and this PR changes things those lanes can't reach:

- **TP head-count validation.** `gemma4_moe/parallelization.py` now checks every per-layer `(num_attention_heads, num_key_value_heads)` against `tp_size` instead of one global pair. Our gemma4 parity test only covers `tp_size=2`, but the shipped recipes run `tp4pp2` and `tp4pp4` — a per-layer count that divides by 2 need not divide by 4.
- **VLM pipeline path.** `hf_utils.py` changes `image_features` from a tensor to a list (`torch.cat(...)`). We cover `pp_size=2` on a single VLM; the `pp8` and `tp4pp4` VLM recipes only run in nemo-ci.
- **Models with no GitHub parallelism coverage.** `nemotron_v3` (~24 changed lines), `qwen3_5_moe/cp_linear_attn.py` (~26), `qwen3_5`, `inkling`. Their PP/CP/EP recipes are nemo-ci only, and CP correctness isn't something unit tests catch.

**Comment by athitten** [comment]
Also I believe this line in PR desc is not relevant: VLM pytest remains unproven: its image attempted an offline setuptools installation and failed before collection ? since the  L2_HF_Transformer_VLM passed in CI here

**Comment by athitten** [comment]
A rebase would pick up the deepseek_v4 PP/EP parity tests from #3613, which exercise MoE under PP.

**Comment by athitten** [comment]
@svcnemo-autobot Could we run the GitLab nemo-ci release suite before merging this?

GitHub CI is green, but it caps at 2 GPUs, and this PR changes things those lanes can't reach:

TP head-count validation. gemma4_moe/parallelization.py now checks every per-layer (num_attention_heads, num_key_value_heads) against tp_size instead of one global pair. Our gemma4 parity test only covers tp_size=2, but the shipped recipes run tp4pp2 and tp4pp4 — a per-layer count that divides by 2 need not divide by 4.
VLM pipeline path. hf_utils.py changes image_features from a tensor to a list (torch.cat(...)). We cover pp_size=2 on a single VLM; the pp8 and tp4pp4 VLM recipes only run in nemo-ci.
Models with no GitHub parallelism coverage. nemotron_v3 (~24 changed lines), qwen3_5_moe/cp_linear_attn.py (~26), qwen3_5, inkling. Their PP/CP/EP recipes are nemo-ci only, and CP correctness isn't something unit tests catch.

**Comment by athitten** [comment]
@svcnemo-autobot Also I believe this line in PR desc is not relevant: VLM pytest remains unproven: its image attempted an offline setuptools installation and failed before collection ? since the L2_HF_Transformer_VLM passed in CI here

**Comment by athitten** [comment]
@svcnemo-autobot A rebase would pick up the deepseek_v4 PP/EP parity tests from https://github.com/NVIDIA-NeMo/Automodel/pull/3613, which exercise MoE under PP.
